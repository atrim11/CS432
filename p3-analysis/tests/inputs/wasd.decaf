int world;
void hello;
bool print;

def int main(int c, int b, int y, bool d, void a, bool f, void e) {

  int now;
  bool now1;
  bool faelse;
  bool help;
  int test;
  continue;
  break;
  now1 = true;
  now1 = xyz;
  test = 2;
  now3 = abcd[hi];
  help = (5 + 3 - 2);
  print_str("hello\\n \"\\\\world");
  test = -0xffff123abcdef;
  test = 123456789;
return 5;
}

def void wow (int a, int b, int c, bool d) {return -99;}

def void bad(int a, int b) {}
