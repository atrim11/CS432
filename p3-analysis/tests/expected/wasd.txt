Void variable 'hello' on line 2
Invalid 'continue' outside loop on line 12
Invalid 'break' outside loop on line 13
Symbol 'xyz' undefined on line 15
Symbol 'now3' undefined on line 17
Symbol 'abcd' undefined on line 17
Symbol 'hi' undefined on line 17
Type mismatch: bool is incompatible with int on line 18
Invalid non-void return from void function on line 25
'main' must take no parameters
